instructure/canvas-ios
# Teacher Bulk Publish Progress screen UI

refs: [MBL-17305](https://instructure.atlassian.net/browse/MBL-17305)
affects: Teacher
release note: none

## What's new
- added Progress screen for All Modules or single Module bulk publish
- added Cancel snackbar
- integrated them into ModuleList

### Details
- added view & viewmodel
- added temporary dummy interactor for QA purposes
- refactored publish action enum to `ModulePublishAction` to include `subject`
- updated `findSnackBarViewModel()` to handle more cases
- unified ProgressViewStyle coloring

## Test plan
- Enable experimental feature flag in Developer Menu
- To reproduce success case: publish/unpublish All Modules (in navBar)
- To reproduce failure case: publish/unpublish a Module (in Module section header)
  - update should fail after a few seconds
- See ticket for expectations

## Screenshots

<table>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/15140172/d1e22968-a2e8-459b-94e9-10c102cff37e" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/15140172/112d163d-8411-4ed2-a9dc-29719fb935f4" maxHeight=500></td>
</tr>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/15140172/618ee676-42f3-450b-a0b3-0998ae15282a" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/15140172/9cc0d986-639b-4be6-97a5-6184f610a02b" maxHeight=500></td>
</tr>
</table>


## Checklist


[MBL-17305]: https://instructure.atlassian.net/browse/MBL-17305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
